app/actions/views/file_upload.js

```diff
@@ -99,7 +99,15 @@ export function retryFileUpload(file, rootId) {
     };
 }
 
-export function handleClearFiles(channelId, rootId) {
+export function handleClearFiles(channelId, rootId, onlyFailed = false) {
+    if (onlyFailed) {
```

Comment:
Can you make this a separate function? I'm not a fan of ones that completely change their behaviour through a flag

Reply:
I wanted to avoid that but sure, I can go ahead and do that now